packages/extract/src/Extract.ts

```diff
@@ -75,7 +76,13 @@ export class Extract implements ISystem
      */
     public base64(target: DisplayObject | RenderTexture, format?: string, quality?: number): string
     {
-        return this.canvas(target).toDataURL(format, quality);
+        const canvas = this.canvas(target);
+
+        if (canvas.toDataURL !== undefined)
+        {
+            return canvas.toDataURL(format, quality);
+        }
+        throw new Error('TODO: base64 for OffscreenCanvas');
```

Comment:
Why not fallback to something like this?
```js
return canvas.convertToBlob().then(blob => new FileReaderSync().readAsDataURL(blob));
```
Since this is for v7, changing the return from `string` to `Promise<string>` is okay with me.

Reply:
I think we should use `URL.createObjectURL(blob)` here, since `FileReaderSync` is only available in worker thread. Using `OffscreenCanvas` in main thread is possible.

Reply:
`FileReader.readAsDataURL` instead of `FileReaderSync.readAsDataURL`, I think.

Reply:
Yeah, my bad, FileReader.

Reply:
Done in de690ddc42a109ec635e59a826f41c1958809a34:
https://github.com/pixijs/pixijs/blob/de690ddc42a109ec635e59a826f41c1958809a34/packages/extract/src/Extract.ts#L77-L99